### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by bczovek**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -40,22 +41,22 @@
         <dependency>
```
Comment (mlozyn) [inline pom.xml:59]: Consider to move bom under dependency management section as here https://learn.microsoft.com/en-us/semantic-kernel/get-started/quick-start-guide?pivots=programming-language-java. Then remove explicit version number from semantic kernel dependencies, pom will handle it for you ...

File: src/main/java/com/epam/training/gen/ai/service/GenAiChatService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        return true;
+    }
+
+    private void addResultsToHistory(List<ChatMessageContent<?>> results) {
+        if(results != null) {
+            for (ChatMessageContent<?> message : results) {
+                chatHistory.addMessage(message);
+            }
+        }
```
Comment (mlozyn) [inline src/main/java/com/epam/training/gen/ai/service/GenAiChatService.java:62]: what about if Objects.nonNull(results) results.forEach(chatHistory::addMessage) ?

File: src/main/java/com/epam/training/gen/ai/service/GenAiChatService.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class GenAiChatService {
+
+    private final ChatCompletionService chatCompletionService;
```
Comment (mlozyn) [inline src/main/java/com/epam/training/gen/ai/service/GenAiChatService.java]: it's not necessary to pass ChatCompletionService here explicitly. you could take it from kernel as here: 

ChatCompletionService chatCompletionService = kernel.getService(ChatCompletionService.class);

https://learn.microsoft.com/en-us/semantic-kernel/concepts/ai-services/chat-completion/?tabs=csharp-AzureOpenAI%2Cpython-AzureOpenAI%2Cjava-AzureOpenAI&pivots=programming-language-java

File: src/main/java/com/epam/training/gen/ai/configuration/GenAiConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        return OpenAIChatCompletion.builder()
+                .withModelId(connectionProperties.deploymentName())
+                .withOpenAIAsyncClient(client)
+                .build();
+    }
+
+    @Bean
+    public ChatHistory chatHistory() {
+        return new ChatHistory();
+    }
```
Comment (mlozyn) [inline src/main/java/com/epam/training/gen/ai/configuration/GenAiConfiguration.java:40]: ideally it should not be a singleton. I would say it's better to instantiate ChatHistory in ChatService ... what if there will be 2 chats ?

Comment (bczovek) [reply src/main/java/com/epam/training/gen/ai/configuration/GenAiConfiguration.java:40]: You are correct. I am going to move away from singleton chat history in later modules. Thank you for your comments!

**Pull request closed without merging**